generics/arraypool/doc.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package arraypool
+
+import (
+	"github.com/cheekybits/genny/generic"
+)
+
+// elemType is the generic type for use with the specialized array pool.
+// NB: we need this type decl to allow codegen and tests to succeed. It's
+// kept in a separate file to allow generation of specialized array pools
+// without requiring a corresponding element declaration.
+type elemType generic.Type
```

Comment:
Hm not sure if I follow, what happens if you put this typedef in `pool.go`?

Reply:
Didn't work with other genny but seems to be fine with the one you suggested we use. Will remove it.

Reply:
done